core/src/main/java/org/apache/accumulo/core/fate/Fate.java

```diff
@@ -88,6 +89,53 @@ public enum TxInfo {
     TX_NAME, AUTO_CLEAN, EXCEPTION, TX_AGEOFF, RETURN_VALUE
   }
 
+  public enum FateOperation {
+    COMMIT_COMPACTION(null),
+    NAMESPACE_CREATE(TFateOperation.NAMESPACE_CREATE),
+    NAMESPACE_DELETE(TFateOperation.NAMESPACE_DELETE),
+    NAMESPACE_RENAME(TFateOperation.NAMESPACE_RENAME),
+    SHUTDOWN_TSERVER(null),
+    SYSTEM_SPLIT(null),
+    TABLE_BULK_IMPORT2(TFateOperation.TABLE_BULK_IMPORT2),
+    TABLE_CANCEL_COMPACT(TFateOperation.TABLE_CANCEL_COMPACT),
+    TABLE_CLONE(TFateOperation.TABLE_CLONE),
+    TABLE_COMPACT(TFateOperation.TABLE_COMPACT),
+    TABLE_CREATE(TFateOperation.TABLE_CREATE),
+    TABLE_DELETE(TFateOperation.TABLE_DELETE),
+    TABLE_DELETE_RANGE(TFateOperation.TABLE_DELETE_RANGE),
+    TABLE_EXPORT(TFateOperation.TABLE_EXPORT),
+    TABLE_IMPORT(TFateOperation.TABLE_IMPORT),
+    TABLE_MERGE(TFateOperation.TABLE_MERGE),
+    TABLE_OFFLINE(TFateOperation.TABLE_OFFLINE),
+    TABLE_ONLINE(TFateOperation.TABLE_ONLINE),
+    TABLE_RENAME(TFateOperation.TABLE_RENAME),
+    TABLE_SPLIT(TFateOperation.TABLE_SPLIT),
+    TABLE_TABLET_AVAILABILITY(TFateOperation.TABLE_TABLET_AVAILABILITY);
+
+    private final TFateOperation top;
+    private static final EnumSet<FateOperation> nonThriftOps =
+        EnumSet.of(COMMIT_COMPACTION, SHUTDOWN_TSERVER, SYSTEM_SPLIT);
+
+    FateOperation(TFateOperation top) {
+      this.top = top;
+    }
+
+    public static FateOperation fromThrift(TFateOperation top) {
+      return FateOperation.valueOf(top.name());
```

Comment:
I think this will only work if the FateOperation and TFateOperation enum names are exactly the same. I wonder if a switch statement would be better.

Reply:
This was intentional. I think it might be confusing/wrong to have a FateOperation with an equivalent TFateOperation but different names.

Reply:
I agree that the names make sense to be the same. If they are not the same then this will produce a run time exception. With a switch statement there might be a compilation warning that not all enum values are handled in the case that new enum elements are added in a future version. This works now, I'm just trying to figure out if we can try to future-proof it. If not, so be it.

Reply:
I'm not quite sure I follow. If we use a switch statement like:
```
switch (top) {
        case TABLE_CREATE:
          return FateOperation.TABLE_CREATE;
        case TABLE_CLONE:
          return FateOperation.TABLE_CLONE;
        ...
```
We would still need a default case, so this doesn't get rid of our runtime exception.
Am I misunderstanding?

Reply:
So, if you don't include a default case, then you will get a warning about not handling all types. Given that all cases return from the method, you could still throw an IllegalArgumentException if the end of the method is reached. Something like:

```
switch (top) {
  case TABLE_CREATE:
    return FateOperation.TABLE_CREATE;
  case TABLE_CLONE:
    return FateOperation.TABLE_CLONE;
  ...
}
throw new IllegalArgumentException("Unhandled type: " + top);
```

Reply:
Interesting. I see a warning in my IDE, not a build time. Feel free to ignore this suggestion then.

Reply:
Hmm yeah maybe different versions of Java? I'm not sure...

Reply:
I'll mark this as resolved.
This should still be pretty future-proof with the new test which would catch any issues like a TFateOperation not having an equivalent FateOperation